lib/warden/test/mock.rb

```diff
@@ -37,7 +37,7 @@ def env
       def app
         @app ||= begin
           opts = {
-            failure_app: lambda {
+            failure_app: lambda { |env|
```

Comment:
Do we need the `env` here?

Reply:
Yes, in newer Ruby, lambda would strictly check the arity. Without this change, I'll need to provide a `failure_app` in the throwing test because it's using this default `failure_app`, which needs to take an `env`.

Reply:
I just realized that I should probably name this `_e` to match the app here. Let me know if I need to rename it.

Reply:
I renamed this while I am at it.